SMLHelper/Options/ModOptionEventAttribute.cs

```diff
@@ -0,0 +1,32 @@
+﻿namespace SMLHelper.V2.Options
+{
+    using Interfaces;
+    using System;
+
+    /// <summary>
+    /// Abstract base attribute used to signify a method to call whenever the derivative event is invoked for the attributed member.
+    /// </summary>
+    /// <remarks>
+    /// The method must be a member of the same class.
+    /// </remarks>
+    [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property | AttributeTargets.Method, AllowMultiple = true)]
+    public abstract class ModOptionEventAttribute : Attribute, IModOptionEventAttribute
```

Comment:
Would limiting these attributes to only be used on strictly Properties (or strictly Fields) help simplify the rest of the backing code?

Reply:
Yes, but also no. I'm moving to `Traverse` for most of the getting/setting, so the backing code becomes automatically super simplified in that regard. I specifically want to allow for properties *and* fields because ConfigFile is based off of Newtonsoft.Json which also allows for both. And that means having to change the member design of your ConfigFile if you already use both in most cases, and I think that's not ideal.

Yes, the code-base for this is big. Of course it is - it's a big task it undertakes. But how "complex" it is is being worked on. Don't jump the gun on wanting to strip out half of it.

Reply:
PS: Side-benefit of moving to `Traverse` for the getting/setting is that it also caches the reflection lookups for us under-the-hood, so that's nice.